Aspen-Discovery/aspen-discovery
# Symphony Contact Information updates

1. Some Symphony libraries store information about how patrons wish to receive notices and billing notices in user category fields.  Which user category number contains this information is arbitrary and varies by library.  
- Added settings in Library Systems -> ILS/Account Integration -> User Profile: User Category Number for Notice Preference, User Category Notice Options, User Category Number for Billing Notice Preference, User Category Billing Notice Options, along with database fields in the Library table (symphonyNoticeCategoryNumber, symphonyNoticeCategoryOptions, symphonyBillingNoticeCategoryNumber, symphonyBillingNoticeCategoryOptions).  
- If a category number and valid options are set up, the Contact Information page will allow the user to see and update their preferences for notice and billing notice delivery.

**Test plan**
1. Add category number and valid notice options to new settings in Library Systems -> ILS/Account Integration -> User Profile for a Symphony library that uses categories this way.
2. Verify that a patron who has this library as their home library now sees notice and billing notice preferences on the Contact Information page and can update them.
3. Verify that a patron who does not have this library as their home library does not see these preferences.
4. Verify that if only notices or only billing notices are set up only the one that is set up shows on Contact Information.
5. Verify that non-Symphony libraries do not have any changes on their Contact Information page, regardless of the new settings.

2. Some Symphony libraries use a different field than PHONE for phone numbers (DAYPHONE or HOMEPHONE) and this caused an error on the Contact Information page (Expected a resource of type '/policy/patronAddress1' for the field 'address1[8]').  
- Added setting for Symphony Default Phone Field in Library Systems -> ILS/Account Integration -> User Profile and added SymphonyDefaultPhoneField to Library table in the database, which defaults to 'PHONE'.  
- The Contact Information page uses SymphonyDefaultPhoneField when updating the phone number.  
- When updating patron data in the Aspen database, if either DAYPHONE or HOMEPHONE exists but PHONE does not, they will be saved in the Aspen phone field.  

**Test Plan**
1. Using a Symphony library that uses PHONE as the default for phone numbers, do not change the new setting and verify that you are able to change a patron's phone number on the Contact Information page without getting an error.  Verify that the number is updated both in the Aspen database and in the ILS.
2. Using a Symphony library that does not use PHONE but uses DAYPHONE or HOMEPHONE, change the new setting to the appropriate field name.  Again verify that you are able to change a patron's phone number on the Contact Information page without getting an error.  Verify that the number is updated both in the Aspen database and in the ILS.
3. Take the same steps with a non-Symphony library and verify that nothing has changed about their Contact Information page form submission.
4. This change may impact Symphony self registration.  Verify that patrons are still able to self-register and include a phone number.

3. Fixed a typo that was appending '70' to the default button color.

**Test Plan**
1. Verify that buttons with the default-btn class are the color set by the theme.
